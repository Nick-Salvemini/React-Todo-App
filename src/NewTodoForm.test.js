import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

describe('NewTodoForm component', () => {
    it('renders without crashing', () => {
        render(<NewTodoForm addTodo={() => { }} />);
    });

    it('matches snapshot', () => {
        const { container } = render(<NewTodoForm addTodo={() => { }} />);
        expect(container).toMatchSnapshot();
    });

    it('calls addTodo function with form data on submission', () => {
        const addTodoMock = jest.fn();
        const { getByLabelText, getByText } = render(<NewTodoForm addTodo={addTodoMock} />);

        fireEvent.change(getByLabelText(/task/i), { target: { value: 'Sample task' } });
        fireEvent.click(getByText(/add new task/i));

        expect(addTodoMock).toHaveBeenCalledWith({ task: 'Sample task' });
    });
});