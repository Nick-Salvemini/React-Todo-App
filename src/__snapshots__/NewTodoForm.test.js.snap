// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm component matches snapshot 1`] = `
<div>
  <form>
    <label
      for="task"
    >
      Task: 
    </label>
    <input
      id="task"
      name="task"
      placeholder="New Task"
      type="text"
      value=""
    />
    <button>
      Add New Task
    </button>
  </form>
</div>
`;
